src/Aspire.Hosting.Azure/Provisioning/Provisioners/AzureContainerAppsInfastructure.cs

```diff
@@ -0,0 +1,725 @@
+                                                    AnyH2 = g.Any(e => e.Transport is "http2")
+                                                })
+                                                .ToList();
+
+        // Failure cases
+
+        // Multiple external endpoints are not supported
+        if (endpointsByContainerPort.Count(g => g.External) > 1)
+        {
+            throw new NotSupportedException("Multiple external endpoints are not supported");
+        }
+
+        // Any external non-http endpoints are not supported
+        if (endpointsByContainerPort.Any(g => g.External && !g.IsHttpOnly))
+        {
+            throw new NotSupportedException("External non-HTTP(s) endpoints are not supported");
+        }
+
+        // Get all http only groups
+        var httpOnlyEndpoints = endpointsByContainerPort.Where(g => g.IsHttpOnly).ToArray();
+
+        // Do we only have one?
+        var httpIngress = httpOnlyEndpoints.Length == 1 ? httpOnlyEndpoints[0] : null;
+
+        if (httpIngress is null)
+        {
+            // We have more than one, pick prefer external one
+            var externalHttp = httpOnlyEndpoints.Where(g => g.External).ToArray();
+
+            if (externalHttp.Length == 1)
+            {
+                httpIngress = externalHttp[0];
+            }
+            else if (httpOnlyEndpoints.Length > 0)
+            {
+                // We have multiple internal only HTTP endpoints, we don't know which one should map to the ingress
+                // REVIEW: We could pick the first one, but that seems strange.
+                throw new NotSupportedException("Multiple internal only HTTP(s) endpoints are not supported.");
+            }
+        }
+
+        if (httpIngress is not null)
+        {
+            // We're processed the http ingress, remove it from the list
+            endpointsByContainerPort.Remove(httpIngress);
+        }
+
+        // ACA can't handle > 5 additional ports so throw if that's the case here
+        if (endpointsByContainerPort.Count > 5)
+        {
+            throw new NotSupportedException("More than 5 additional ports are not supported. See https://learn.microsoft.com/en-us/azure/container-apps/ingress-overview#tcp for more details.");
+        }
+
+        // Now we map the remainig endpoints. These should be internal only tcp/http based endpoints
+
+        var sb = new StringBuilder();
+
+        sb.AppendLine("ingress: {");
+        if (httpIngress is not null)
+        {
+            sb.AppendLine(CultureInfo.InvariantCulture, $"  external: {httpIngress.External.ToString().ToLowerInvariant()}");
+            sb.AppendLine(CultureInfo.InvariantCulture, $"  targetPort: {httpIngress.Port ?? (resource is ProjectResource ? 8080 : 80)}");
+            sb.AppendLine(CultureInfo.InvariantCulture, $"  transport: '{(httpIngress.AnyH2 ? "http2" : "http")}'");
+        }
+        else
+        {
+            var port = endpointsByContainerPort[0].Port;
```

Comment:
So this is taking the first non HTTP endpoint and promoting it to be the tcp ingress?

Reply:
Yep